components/dynamic/SummitData/Weather/Current/index.js

```diff
@@ -0,0 +1,47 @@
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useWeatherUnit } from "@/contexts/WeatherUnit";
+import { useSummitData } from "@/contexts/SummitData";
+import Loader from "@/components/atomic/Loader";
+import WidgetSection from "@/components/layout/WidgetSection";
+import TemperatureCurrent from "@/components/widgets/CurrentData/patterns/Temperature";
+import WindspeedCurrent from "@/components/widgets/CurrentData/patterns/Windspeed";
+import PrecipitationCurrent from "@/components/widgets/CurrentData/patterns/Precipitation";
+import { convertTemperature, convertWindspeed } from "@/helpers/converters";
+
+const CurrentWeather = () => {
+  const { t } = useTranslation();
+  const [isOpen, setOpen] = useState(true);
+  const [{ tempUnit, windspeedUnit }] = useWeatherUnit();
+  const { currentData, loading = true } = useSummitData();
+  const sectionProps = {
+    title: t("summit_dashboard.sections.weather.current"),
+    onToggleCallback: (value) => setOpen(value),
+    isOpen,
+  };
+
+  if (loading && !currentData)
+    return (
+      <WidgetSection {...sectionProps}>
+        <Loader isVisible={true} />
+      </WidgetSection>
+    );
+
+  const { temperature0, windspeed, relativeHumidity } = currentData;
+  const temperature = convertTemperature(temperature0, tempUnit);
```

Comment:
each of the sections of the weather widgets has it's own container: current, hourly, and daily data. They are roughly the same, each pulls out it's data from the `useSummitData()` hook and then puts it into components.

This brings me to a point I want to discuss. Should the summit data provider handle the pre-processing for the data it retrieves from the EFD? Right now it just stores the raw values and the widget containers handle processing which is a combo of morphing into objects that the display components recognize and converting to selected temperature/windspeed units.

I think converting to end user temp/wind units is over-concerned and complicated, but what about stuff like setting up the objects themselves? Here's a few things I think could be pre-processed:

- mapping to recognizable properties (temperature0 -> temperature, speed_mean -> windspeed, direction_mean -> direction)
- slicing off extra data, for example if you query backwards 7 days you end up with 8 results because it also queries today from midnight -> present.
- putting timestamps into date objects